### ISSUE TITLE
Hyperlink escape sequence causes bad output in Ubuntu 16 LTS terminal

### ISSUE THREAD

**Issue by jay** [curl/curl#10428]
### I did this

~~~
curl --head https://www.google.com
~~~

In Ubuntu 16 LTS terminal the location header has unrecognized escape sequences and duplicate URL.
~~~
location:]8;;https://www.google.com/\ https://www.google.com/
]8;;\content-type: text/html; charset=UTF-8
~~~

![Capture](https://user-images.githubusercontent.com/965580/217159919-ff897731-71a0-4a79-b837-e7a2562b4a67.PNG)

/cc @dfandrich

**Comment by dfandrich** [comment]
What terminal emulator does this use? On all the ones I tried, the escape
sequence is either interpreted correctly or ignored and not displayed.

**Comment by jay** [comment]
Gnome Terminal 3.18.3 + VTE 0.42.5

**Comment by bagder** [comment]
@dfandrich is this only because of an older gnome terminal and we can just ask users to run a later version or is there anything more clever we can here?

**Comment by dfandrich** [comment]
OSC 8 support was added to Gnome Terminal 6 years ago in ver. 3.25.1. There's a list tracking support in terminal emulators [on this page](/Alhadis/OSC8-Adoption/). The [OSC 8 spec](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda#backward-compatibility) also says

>  Any terminal that correctly implements OSC parsing according to ECMA-48 is guaranteed not to suffer from compatibility issues. That is, even if explicit hyperlinks aren't supported, the target URI is silently ignored and the supposed-to-be-visible text is displayed, without artifacts.

It sounds like there may have been a bug in Gnome Terminal that didn't do this, but other terminal emulators, even those that don't support OSC 8, should ignore the extra data and that is what I found in my testing. That spec page lists four terminal emulators that still didn't work properly as of 3 years ago, but all of them are working (mostly) today in their latest versions.

I can't think how  we could easily and reliably detect whether a terminal emulator is buggy or not and dynamically enable this. xterm-style answerback strings might theoretically do it if those were still a thing, otherwise it would take linking with an X11 library  and even that wouldn't work all the time. My suggestion would be to just use --no-styled-output on those few cases where it doesn't work, or better yet, upgrade your terminal emulator since there aren't any known buggy implementations left according to that page.
